en/docs/insights/choreo-insights-api.md

```diff
@@ -0,0 +1,4214 @@
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">timeFilter</td>
+<td valign="top"><a href="#timefilter">TimeFilter</a>!</td>
+<td>
+
+Time filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">deviceFilter</td>
+<td valign="top"><a href="#devicefilter">DeviceFilter</a></td>
+<td>
+
+Device filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">dataFilter</td>
+<td valign="top"><a href="#datafilter">DataFilter</a>!</td>
+<td>
+
+Data filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>getTopUserAgents</strong></td>
+<td valign="top">[<a href="#useragent">UserAgent</a>]</td>
+<td>
+
+Returns a list of top user agent by hit count within the specified time range for the given combination of tenant
+ID, environment ID, and organization ID. The result list will include top 9 user agents with the respective hit
+count and all the other user agents labelled as 'Other' with the respective cumulated hit count. Optionally, you can
+configure the `deviceFilter` parameter to further filter the results.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">timeFilter</td>
+<td valign="top"><a href="#timefilter">TimeFilter</a>!</td>
+<td>
+
+Time filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">deviceFilter</td>
+<td valign="top"><a href="#devicefilter">DeviceFilter</a></td>
+<td>
+
+Device filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">dataFilter</td>
+<td valign="top"><a href="#datafilter">DataFilter</a>!</td>
+<td>
+
+Data filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>getAPIUsageByGeoLocation</strong></td>
+<td valign="top"><a href="#apiusagebygeolocation">APIUsageByGeoLocation</a></td>
+<td>
+
+NOTE: The data required for this operation is not yet collected in the database. Hence, this operation will return
+empty results.
+
+Returns API usage by country within the specified time range for the given combination of tenant ID, environment ID,
+and organization ID. Optionally, you can configure the `geoLocationFilter` parameter to further filter the results.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">timeFilter</td>
+<td valign="top"><a href="#timefilter">TimeFilter</a>!</td>
+<td>
+
+Time filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">geoLocationFilter</td>
+<td valign="top"><a href="#geolocationfilter">GeoLocationFilter</a></td>
+<td>
+
+Geo location filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" align="right" valign="top">dataFilter</td>
+<td valign="top"><a href="#datafilter">DataFilter</a>!</td>
+<td>
+
+Data filter
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>getAPIsUsageByApplications</strong></td>
+<td valign="top">[<a href="#apiusagebyapplication">APIUsageByApplication</a>]</td>
+<td>
+
+Returns a list of each API usage by each application within the specified time range for the given combination of
```

Comment:
Shall we replace "Returns a list of each API usage by each application" with "Returns usage details of each API by application..."?